lib/server.coffee

```diff
@@ -40,32 +53,62 @@ getPagePath = (name) ->
 #   console.log(token)
 ###
 exports.awaitForToken = (options) ->
-	app = express()
-	app.use bodyParser.urlencoded
-		extended: true
-
-	server = app.listen(options.port)
+	{ app, server } = createServer(port: options.port)
 
 	return new Promise (resolve, reject) ->
-		app.post options.path, (request, response) ->
-			token = request.body.token
+		closeServer = (errorMessage, successPayload) ->
+			server.close ->
+				if errorMessage
+					reject(new Error(errorMessage))
+					return
+
+				resolve(successPayload)
 
-			utils.isTokenValid(token)
-			.then (isValid) ->
-				if isValid
-					return response.status(200).sendFile getPagePath('success'), ->
-						request.connection.destroy()
-						server.close ->
-							return resolve(token)
+		renderAndDone = ({ request, response, viewName, errorMessage, statusCode, token }) ->
+			return getContext(viewName)
+			.then (context) ->
+				response.status(statusCode || 200).render(viewName, context)
+				request.connection.destroy()
+				closeServer(errorMessage, token)
 
-				throw new Error('No token')
+		app.post options.path, (request, response) ->
+			token = request.body.token?.trim()
+
+			Promise.try ->
+				if not token
+					throw new Error('No token')
+				return utils.isTokenValid(token)
+			.tap (isValid) ->
+				if not isValid
+					throw new Error('Invalid token')
+			.then ->
+				renderAndDone({ request, response, viewName: 'success', token })
 			.catch (error) ->
-				response.status(401).sendFile getPagePath('error'), ->
-					request.connection.destroy()
-					server.close ->
-						return reject(new Error(error.message))
+				renderAndDone({
+					request, response, viewName: 'error',
+					statusCode: 401, errorMessage: error.message
+				})
 
 		app.use (request, response) ->
 			response.status(404).send('Not found')
-			server.close()
-			return reject(new Error('No token'))
+			closeServer('Unknown path or verb')
+
+exports.getContext = getContext = (viewName) ->
+	if viewName is 'success'
+		return Promise.props
+			dashboardUrl: resin.settings.get('dashboardUrl')
```

Comment:
Yes, that's the whole point of the settings module (also it's already used in this resin-cli-auth module for the actual redirect URL generation, check utils)